divi/src/CoinMinter.cpp

```diff
@@ -350,3 +350,8 @@ void CoinMinter::addExtraTransaction(const CTransaction& tx)
 {
     extraTransactions_.push_back(tx);
 }
+
+void CoinMinter::useCustomCoinstakeForNextBlock(const CTransaction& tx)
+{
+    blockFactory_->UseCustomCoinstakeForNextBlock(tx);
+}
```

Comment:
Similar to the other comment about not adding to the class directly, you could modify the RegTestBlockFactory to use a custom coinstake when creating a new block. Perhaps by making the blockFactory_ variable a protected member